aiosip/peers.py

```diff
@@ -268,27 +266,19 @@ def _create_peer(self, peer_addr):
         return await self._dispatch(protocol, peer_addr)
 
     def connection_lost(self, protocol):
-        peer_addr = protocol.transport.get_extra_info('peername')
-        local_addr = protocol.transport.get_extra_info('sockname')
-        peer = self._peers.pop((peer_addr, local_addr), None)
-        if peer:
-            peer.close()
+        for key, peer in list(self._peers.items()):
+            if peer._protocol == protocol:
+                peer._disconnected()
+                self._peers.pop(key)
```

Comment:
We probably also want to delete the protocol from `self._protocols` here.

Reply:
Hm. That's true. To be honest I'm not quite comfortable about all the back & forth we do on self._protocols and self._peers, it makes a lot of assumptions about the underlying transport.

I there a 1 <-> 1 relationship between peer and protocol?

Reply:
There probably is right now, but shouldn't be. TCP+UDP should be a valid setup, no?